src/main/java/org/mockito/plugins/MockMaker.java

```diff
@@ -101,6 +101,16 @@ void resetMock(
             MockCreationSettings settings
     );
 
+    /**
+     * Cleans up internal state for specified {@code mock}. You may assume there won't be any interaction to this {@code mock}
+     * after this is called.
+     *
+     * @param mock the mock instance whose internal state is to be cleaned.
+     * @since 2.24.4
+     */
+    @Incubating
+    void cleanUpMock(Object mock);
```

Comment:
Sadly this is a breaking change. Moreover, not all mockmakers actually need to clean up. I think it is better to have a separate interface for this and do an `instanceof` check instead.

Reply:
Done.

Not a significant change at all.